### PR TITLE
8315871: Opensource five more Swing regression tests

### PR DESCRIPTION
**Pull request by azuev-java**
Clean up and move to open source five tests.
javax/swing/AncestorNotifier/4817630/bug4817630.java
javax/swing/BoxLayout/4191948/bug4191948.java
javax/swing/ComponentInputMap/4248723/bug4248723.java
javax/swing/DefaultBoundedRangeModel/4297953/bug4297953.java
javax/swing/DefaultButtonModel/4097723/bug4097723.java

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315871](https://bugs.openjdk.org/browse/JDK-8315871): Opensource five more Swing regression tests (**Bug** - P4)

### Reviewers
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer) ⚠️ Review applies to [d9d32378](https://git.openjdk.org/jdk/pull/15702/files/d9d3237871176539e96f7a00055256f49f0ca38f)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15702/head:pull/15702` \
`$ git checkout pull/15702`

Update a local copy of the PR: \
`$ git checkout pull/15702` \
`$ git pull https://git.openjdk.org/jdk.git pull/15702/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15702`

View PR using the GUI difftool: \
`$ git pr show -t 15702`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15702.diff">https://git.openjdk.org/jdk/pull/15702.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15702#issuecomment-1717222866)

### CODE REVIEW

File: test/jdk/javax/swing/DefaultBoundedRangeModel/4297953/bug4297953.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * @bug 4297953
+ * @summary Tests that DefaultBoundedRangeModel doesn't zero out the
+ *          extent value when maximum changes
+ * @run main bug4297953
+ */
+
+import javax.swing.JScrollBar;
+
+public class bug4297953 {
+    public static void main(String[] args)  {
```
Comment (turbanoff) [inline test/jdk/javax/swing/DefaultBoundedRangeModel/4297953/bug4297953.java:35]:
```suggestion
    public static void main(String[] args) {
```

File: test/jdk/javax/swing/BoxLayout/4191948/bug4191948.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            int foo2Width = foo2.getPreferredSize().width;
+            int foo3Width = foo3.getPreferredSize().width;
+            if (totalWidth >= (foo1Width + foo2Width + foo3Width)) {
+                throw new RuntimeException("Panel is too wide");
+            }
+        } finally {
+            if (frame != null) {
+                frame.setVisible(false);
+                frame.dispose();
+            }
```
Comment (DamonGuy) [inline test/jdk/javax/swing/BoxLayout/4191948/bug4191948.java:76]: Should this if statement be run on EDT?

File: test/jdk/javax/swing/AncestorNotifier/4817630/bug4817630.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        bug4817630 test = new bug4817630();
+        try {
+            SwingUtilities.invokeAndWait(test::init);
+            test.start();
+        } finally {
+            test.destroy();
```
Comment (DamonGuy) [inline test/jdk/javax/swing/AncestorNotifier/4817630/bug4817630.java:112]: Either destroy() or the if statement in destroy() for disposing the frame should be on EDT

**Review by DamonGuy — APPROVED** [review]
LGTM. Just move the frame dispose code blocks to EDT.

**Comment by azuev-java** [comment]
Fixed found problems.

File: test/jdk/javax/swing/BoxLayout/4191948/bug4191948.java (hunk 2 above)
Comment (azuev-java) [reply test/jdk/javax/swing/BoxLayout/4191948/bug4191948.java:76]: In new version the entire method is being run on EDT.

File: test/jdk/javax/swing/AncestorNotifier/4817630/bug4817630.java (hunk 3 above)
Comment (azuev-java) [reply test/jdk/javax/swing/AncestorNotifier/4817630/bug4817630.java]: Fixed.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**